lyrebird/event.py

```diff
@@ -121,5 +124,14 @@ def unregister(self, event_bus):
             event_bus.unsubscribe(listener['channel'], listener['func'])
 
     def publish(self, channel, message, *args, **kwargs):
-        context.application.event_bus.publish(channel, message, *args, **kwargs)
-    
\ No newline at end of file
+        application.server['event'].publish(channel, message, *args, **kwargs)
+
+    def alert(self, channel, message, *args, **kwargs):
+        stack = inspect.stack()
+        caller_file = stack[1].filename
+        self.report_file = caller_file[caller_file.rfind('/') + 1:]
+        self.report_method = stack[1].function
+        if isinstance(message, dict):
+            message['report_file'] = self.report_file
+            message['report_method'] = self.report_method
```

Comment:
And could we name it as "function_name" ?

Reply:
Good idea, will accept it and change the name